### ISSUE TITLE
Edit and View Mode For Data Entry Forms

### ISSUE THREAD

**Issue by Patowhiz** [opencdms/Climsoft#558]
As mentioned in the April's 2019 workshop, there is a need to have a feature that helps the user know if he is editing an existing record or creating a new one. 

Currently when a user selects a new value in the key controls, the form changes the records or navigates to a new record if the record does not exist. This inhibits a user from updating values of the fields that form the composite keys.

The proposition is , in edit mode, the key controls won't prompt navigation to new records, and a user will be able to edit its values, the only way to go to a new record is through the add new.

**Comment by mhabimana** [comment]
@Patowhiz : I am trying to understand this.Yes,you are right,it is not possible to edit elements which form composite keys at the form,it is possible to edit and then click the update to update the existing records by the new records,in the current setting to change the composite keys element,you need to select the view button and change the composite keys in the edit mode.

**Comment by Patowhiz** [comment]
Yes, currently you can only change the key field in the datagrid view, which you access when you click the View button.
I'm thinking a user might want to just edit it in the form itself. Assuming I have enter records for the form2daily then I realise I selected the wrong station, as a user I would want to change the station . In this case if I'm on edit mode then changes to the station control won't go to another record.

**Comment by mhabimana** [comment]
@Patowhiz :Sounds like a good idea.

**Comment by maxwellfundi** [comment]
I agree it is easier to have the user edit everything on the form and saves a bit of time  for the user. Additionally, there is a huge chance of error whilst typing on the datagrid- its possible that a station ID like - 61008906 could be 61O08906 which could actually cause a bug. 

However we need the user to be able to know the different modes here. I wonder if we have a switch on the forms, say a checkbox etc to make sure they are aware they are on edit mode rather than entry mode then as soon as the Add New is clicked the form goes to the entry mode.

**Comment by mhabimana** [comment]
@Patowhiz @maxwellfundi : Sounds very interesting.
